satellite/peer.go

```diff
@@ -491,6 +491,21 @@ func ignoreCancel(err error) error {
 	return err
 }
 
+func (peer *Peer) restartLoop(ctx context.Context, name string, service func(context.Context) error) error {
+	for {
+		err := func() (err error) {
+			defer func() {
+				if rec := recover(); rec != nil {
+					err = errs.New("caught panic in service %s: %+v", name, rec)
+				}
+			}()
+			return service(ctx)
+		}()
+		peer.Log.Sugar().Errorf("Service %s failure: %+v", name, err)
+		time.Sleep(time.Second)
```

Comment:
I think it's important to have a storage node restart instead of immediately exiting the process when it can't get a connection, e.g. this bug that @brimstone saw occurring occasionally: https://storjlabs.atlassian.net/browse/V3-1370

Reply:
oh no 1597 was closed! i'll add a random sleep